app/react/Map/specs/Map.spec.js

```diff
@@ -218,6 +224,15 @@ describe('Map', () => {
       callback({ autoCentered: true });
       expect(instance.setViweport).toHaveBeenCalled();
     });
+
+    it('should read and set the map tiler key in the style', () => {
+      const expectedKey = 'abc';
+      settingsAPI.get.mockResolvedValue({ mapTilesKey: expectedKey });
+    });
+
+    it('prepare mapStyle from settings', () => {
+      expect(settingsAPI.get).toBeCalled();
+    });
```

Comment:
I take it this is a Work in Progress.  This tests really don't prove much.  The first one doesn't test anything at all, the second one is testing that the API got called, but nothing else.  Please, don't make this an implementation test, try to test that the componentDiDMount is actually achieving the results you want, and not that the implementation is calling this or other API endpoint.

Reply:
I'm sorry, the first test really doesn't have to be there :(
The second test, however, I did this intentionally, because I think I only have to check that the method that contains the logic is called because the replaceKeysMapStyleJson method has already a set of tests in its own context.
I think you are true and the test as it was written doesn't explain anything about the expected behavior, so, I've changed this one for a pretty small check for the expected changes.